rfd/0235-desktop-access-linux.md

```diff
@@ -0,0 +1,197 @@
+    |
++---|------------------------------+
+|   v                              |
+| +-------------------------+      |
+| |  linux_desktop_service  |      |
+| +-------------------------+      |
+|     ^                 |          |
+|     | X11             | starts   |
+|     v                 v          |
+| +--------+  X11  +-------------+ |
+| |  Xvfb  |<----->|   Desktop   | |
+| +--------+       | Environment | |
+|                  + ------------+ |
+|                                  |
+|                       Linux host |
++----------------------------------+
+```
+
+### UX
+
+#### Setup/discovery
+
+Setup will require creating new node or updating existing one with new `linux_desktop` role and `linux_desktop_service` 
+enabled (it will be disabled by default).
+
+To help with onboarding, there will be new tile in `Enroll a New Resource` view for Linux desktops.
+For the first release it will lead to documentation. Fully fledged guided installation, similar to enrolling SSH nodes, 
+will be added later.
+
+#### User logging in
+
+Linux Desktop Service will create `LinuxDesktop` resource with its UUID as name. They will be presented in the UI as
+one of unified resources, the same way we present Windows desktops. 
+
+After user selects username to log in, UI will redirect to `/web/cluster/:cluster/linux_desktops/:uuid/:login`.
+User will be presented with list of available xsessions (i.e. desktop environments). After selecting one of the items, 
+user will be redirected to `/web/cluster/:cluster/linux_desktops/:uuid/:login/:xsession`. This will enable user to 
+create bookmarks both to session selection screen and to specific xsession. 
+
+If there's only one entry in `/usr/share/xsessions` selection screen will be skipped.
+
+Started session will reuse visual components used for Windows desktop sessions to make the experience consistent.
+
+### Xsessions/desktop environments
+
+List of xsessions available on the target machine will be obtained by listing files in `/usr/share/xsessions`.
+This is the same mechanism that is used by different display managers, like GDM and LightDM, to populate their 
+list of xsessions. 
+
+On connection, Linux Desktop Service will read selected desktop entry file to get command to execute stored in `Exec=`
+section. It will then find free display number, start `Xvfb` with requested screen size and run command obtained 
+earlier as user requested by UI. It will use X11 protocol to interact with `Xvfb`. 
+
+Only X11 environments will be supported, there will be no support for Wayland. Tested desktop environments should
+include at least Xfce, KDE Plasma and GNOME 48 (that's the last version that supports X11).
+
+Desktop environment will be started using similar mechanism as starting shell for SSH access (re-executing `teleport` 
+binary)
+
+### Authentication
+
+On connection, Linux Desktop Service will verify user using mTLS certificate. No other authentication is required.
```

Comment:
Which CA issues the user certificate trusted by the Linux Desktop Service?

Have we considered introducing a new "LinuxDesktopCA" for this feature?

I'm commenting this for 2 reasons:

1. The trust chains for the "Windows CA" (trusted by Windows hosts) and the LinuxDesktopCA (trusted by linux_desktop_service) are inherently different.
2. I'm working on the User/Windows CA split, and splitting has some uncomfortable corner cases on version downgrades and old/new service compatibility.